### PR TITLE
src/client: partially rewrite to more idiomatic Rust

### PR DESCRIPTION
**Pull request by orangecms**
This also shortens the code and reduces some seeming boilerplate.
Turns out that many expressions can be written in a shorter, simpler way. :-)

### CODE REVIEW

**Review by rminnich — APPROVED** [review]
This is great, the irony here is that this is sample code from the ssh package :-)

File: src/client/mod.rs [hunk 1]
```diff
@@ -82,47 +82,30 @@ fn read_config(p: &Path) -> SshConfig {
...
-    };
-    let port = match params.port {
-        None => 22,
-        Some(p) => p,
-    };
-    let host = match host.contains(':') {
-        true => host.to_string(),
-        false => format!("{}:{}", host, port),
+    let host = params.host_name.as_deref().unwrap_or(host);
+    let port = params.port.unwrap_or(22);
```
Comment (rminnich) [inline src/client/mod.rs]: oh, oops, can you change this to 23.

cpu uses the old telnet port.

I wonder if I should not ask ICANN for it :-)

Or should we use the Plan 9 standard -- tcp=ncpu port=17010

?

File: src/client/mod.rs [hunk 2]
```diff
@@ -82,47 +82,30 @@ fn read_config(p: &Path) -> SshConfig {
...
+
+    let formatted_host : String;
+    let host: &str = if host.contains(':') {
```
Comment (rminnich) [inline src/client/mod.rs:89]: is this backwards? Don't you want if !host.contains?

**Review by rminnich — CHANGES_REQUESTED** [review]

File: src/client/mod.rs (hunk 1 above)
Comment (orangecms) [reply src/client/mod.rs]: Either is fine by me. I like the 23, and we could always make this a parameter, CPU_PORT env var or whatever...
ICANN sounds fun 😹

File: src/client/mod.rs (hunk 2 above)
Comment (orangecms) [reply src/client/mod.rs:89]: Yea this is Rust's notion of a ternary expression, a little back-and-forth reading
sorta `host.contains(':') ? host : ...` in other languages

Comment (rminnich) [reply src/client/mod.rs:89]: oh nvm for a second I thought it was a split but it's not. Got it.

File: src/client/mod.rs (hunk 1 above)
Comment (rminnich) [reply src/client/mod.rs]: let's go with 23 for now. That matches the Go code.

Comment (orangecms) [reply src/client/mod.rs]: ok done

**Review by rminnich — APPROVED** [review]

**Pull request merged**